### ISSUE TITLE
Filter expired discount codes

### ISSUE THREAD

**Issue by hor-net** [easydigitaldownloads/easy-digital-downloads#8573]
Hi I create a lot of discount codes with an expiration date, i would be very handy if i could display only the expired codes like i can with the inactive ones and i could bulk delete it.

thank you
Saverio

**Comment by cklosowski** [comment]
Thanks for the feedback. I will make one comment, bulk deleting discount codes that are expired is something you should be careful with. If a discount code is ever used, it shouldn't be deleted, as there is an accounting paper trail that links a discount code to a payment. In proper practices, you should never delete a discount code that has been used, even if it is expired.

**Comment by hor-net** [comment]
You re right, but in case i have thousands os expired discount codes that are never been used how can i clean up the database?
The case here is that i send automatically discount codes after the first purchase to every customer, we make about 2K sales every month so you can imagine that they pile up pretty fast...
